packages/api/api.ts

```diff
@@ -0,0 +1,76 @@
+import getUserContext, { UnchainedServerUserContext } from './user-context';
+import getLocaleContext, {
+  UnchainedServerLocaleContext,
+} from './locale-context';
+
+import createGraphQLServer from './createGraphQLServer';
+import createBulkImportServer from './createBulkImportServer';
+import { configureRoles } from './roles';
+
+import hashPassword from './hashPassword';
+import getCart from './getCart';
+import evaluateContext from './evaluateContext';
+import filterContext from './filterContext';
+import instantiateLoaders, { UnchainedServerLoaders } from './loaders';
+
+export { hashPassword, getCart, evaluateContext, filterContext };
+export * as roles from './roles';
+export * as acl from './acl';
+export * as errors from './errors';
+
+export interface UnchainedAPI {
+  version: string;
+}
+
+export type UnchainedServerContext = UnchainedServerLocaleContext &
+  UnchainedServerUserContext &
+  UnchainedServerLoaders &
+  UnchainedAPI;
+
+export interface UnchainedServerOptions {
+  unchained: UnchainedAPI;
+  rolesOptions: any;
+}
+
+const UNCHAINED_API_VERSION = '0.55.4'; // eslint-disable-line
+
+export const createContextResolver = (unchained: UnchainedAPI) => async ({
+  req,
+}): Promise<UnchainedServerContext> => {
+  const loaders = await instantiateLoaders(req, unchained);
+  // const intermediateContext: Partial<UnchainedServerContext> = {
+  //   ...unchained,
+  //   ...loaders,
+  // };
+  const userContext = await getUserContext(req /* intermediateContext */);
+  const localeContext = await getLocaleContext(req);
+  return {
+    ...unchained,
```

Comment:
Hmm, I'm not sure about this. It's a bit a lot of magic-merges. This can lead to problems if for example `loaders` also has a key `modules`. Why not making just:

```ts
return {
  unchained,
  loaders,
  user: userContext,
  locale: localeContext
  // version is part of `unchained`
}

Reply:
Also thought about it but then If I change this part I'll break all explicit and implicit (helper functions) resolvers so I didn't do it for now.

Reply:
This will be so much easier to refactor once we have all resolvers and fields explicit and typed

Reply:
Ok, that makes a lot of sense.